eth/contracts/some-logic-contract.sol

```diff
@@ -3,17 +3,48 @@
 pragma solidity >=0.6.0 <=0.7.3;
 pragma experimental ABIEncoderV2;
 
+import "./optimistic-roll-in-compatible.sol";
+
 // TODO: Build a Logic Proxy that checks and embeds the version of the state, and directs to other contracts
 
-contract Some_Logic_Contract {
-  // User (address) is a mandatory first field
-  function initialize_state(address user) external payable returns (bytes32) {
+contract Some_Logic_Contract is Optimistic_Roll_In_Compatible {
+  event Impure_Transition(bytes32 data);
+
+  function get_selector(bytes calldata call_data) internal pure returns (bytes4) {
+    return call_data[0] | (bytes4(call_data[1]) >> 8) | (bytes4(call_data[2]) >> 16) | (bytes4(call_data[3]) >> 24);
+  }
+
+  function initialize_state(address user) external payable override returns (bytes32) {
     require(msg.value >= 500000000000000000, "INSUFFICIENT_DEPOSIT");
     return keccak256(abi.encodePacked(user));
   }
 
-  // User (address) is a mandatory first field
-  // Current State (bytes32) is a mandatory second field
+  function optimistic_entry_point(bytes calldata call_data) external view override returns (bytes32 new_state) {
+    bytes4 selector = get_selector(call_data);
+    require(selector == 0xef6f6a42, "INVALID_OPTIMISTIC_CALL");
```

Comment:
I don't think you need to check for the selector. Just have a fallback function that always throws should be good enough.